tempodb/backend/azure/azure.go

```diff
@@ -373,42 +373,42 @@ func (rw *readerWriter) readRange(ctx context.Context, name string, offset int64
 	return nil
 }
 
-func (rw *readerWriter) readAll(ctx context.Context, name string) ([]byte, blob.ETag, error) {
-	blobURL := rw.hedgedContainerURL.NewBlockBlobURL(name)
+func (rw *readerWriter) readAll(ctx context.Context, name string) ([]byte, azcore.ETag, error) {
+	blobClient := rw.hedgedContainerClient.NewBlockBlobClient(name)
 
-	var props *blob.BlobGetPropertiesResponse
-	props, err := blobURL.GetProperties(ctx, blob.BlobAccessConditions{}, blob.ClientProvidedKeyOptions{})
+	props, err := blobClient.GetProperties(ctx, &blob.GetPropertiesOptions{})
 	if err != nil {
 		return nil, "", err
 	}
 
-	destBuffer := make([]byte, props.ContentLength())
+	destBuffer := make([]byte, *props.ContentLength)
 
-	if err := blob.DownloadBlobToBuffer(context.Background(), blobURL.BlobURL, 0, props.ContentLength(),
-		destBuffer, blob.DownloadFromBlobOptions{
-			BlockSize:   blob.BlobDefaultDownloadBlockSize,
-			Parallelism: uint16(maxParallelism),
-			Progress:    nil,
-			RetryReaderOptionsPerBlock: blob.RetryReaderOptions{
-				MaxRetryRequests: maxRetries,
-			},
+	if _, err := blobClient.DownloadBuffer(context.Background(), destBuffer, &blob.DownloadBufferOptions{
+		Range: blob.HTTPRange{
+			Offset: 0,
+			Count:  *props.ContentLength,
 		},
-	); err != nil {
+		BlockSize:   blob.DefaultDownloadBlockSize,
+		Concurrency: maxParallelism,
+		RetryReaderOptionsPerBlock: blob.RetryReaderOptions{
+			MaxRetries: maxRetries,
+		},
+	}); err != nil {
 		return nil, "", err
 	}
 
-	return destBuffer, props.ETag(), nil
+	var etag azcore.ETag
+	if props.ETag != nil {
+		etag = *props.ETag
+	}
+
+	return destBuffer, etag, nil
 }
 
 func readError(err error) error {
-	var storageError blob.StorageError
-	errors.As(err, &storageError)
-
-	if storageError == nil {
-		return errors.Wrap(err, "reading storage container")
-	}
-	if storageError.ServiceCode() == blob.ServiceCodeBlobNotFound {
+	if bloberror.HasCode(err, bloberror.BlobNotFound) {
```

Comment:
what is our confidence in this change? we have a fair amount of functionality dependent on correctly returning `backend.ErrDoesNotExist`

Reply:
The "new" functionality should be identical to the old. `bloberror.HasCode` is a shorthand for checking:
1. Is the response `azcore.ResponseError`.
2. Does the response contain error code `BlobNotFound`.

Additionally, the `TestReadError` test is still passing even with only changing `blob.ServiceCodeBlobNotFound` to `bloberror.BlobNotFound`.